pkg/notify/slack.go

```diff
@@ -40,53 +40,38 @@ var attachmentColor = map[events.Level]string{
 
 // Slack contains Token for authentication with slack and Channel name to send notification to
 type Slack struct {
-	Token     string
-	Channel   string
-	NotifType config.NotifType
-	SlackURL  string // Useful only for testing
+	Token          string
+	Accessbindings []config.Accessbinding
+	NotifType      config.NotifType
+	SlackURL       string // Useful only for testing
 }
 
 // NewSlack returns new Slack object
 func NewSlack(c *config.Config) Notifier {
 	return &Slack{
-		Token:     c.Communications.Slack.Token,
-		Channel:   c.Communications.Slack.Channel,
-		NotifType: c.Communications.Slack.NotifType,
+		Token:          c.Communications.Slack.Token,
+		Accessbindings: c.Communications.Slack.Accessbindings,
+		NotifType:      c.Communications.Slack.NotifType,
 	}
 }
 
 // SendEvent sends event notification to slack
 func (s *Slack) SendEvent(event events.Event) error {
-	log.Debug(fmt.Sprintf(">> Sending to slack: %+v", event))
-
+	log.Debug(fmt.Sprintf(">> Sending event to slack: %+v", event))
 	api := slack.New(s.Token)
 	if len(s.SlackURL) != 0 {
 		api = slack.New(s.Token, slack.OptionAPIURL(s.SlackURL))
 	}
 	attachment := formatSlackMessage(event, s.NotifType)
 
-	// non empty value in event.channel demands redirection of events to a different channel
-	if event.Channel != "" {
-		channelID, timestamp, err := api.PostMessage(event.Channel, slack.MsgOptionAttachments(attachment), slack.MsgOptionAsUser(true))
-		if err != nil {
-			log.Errorf("Error in sending slack message %s", err.Error())
-			// send error message to default channel
-			if err.Error() == "channel_not_found" {
-				msg := fmt.Sprintf("Unable to send message to Channel `%s`: `%s`\n```add Botkube app to the Channel %s\nMissed events follows below:```", event.Channel, err.Error(), event.Channel)
-				go s.SendMessage(msg)
-				// sending missed event to default channel
-				// reset event.Channel and send event
-				event.Channel = ""
-				go s.SendEvent(event)
-			}
-			return err
-		}
-		log.Debugf("Event successfully sent to channel %s at %s", channelID, timestamp)
-	} else {
-		// empty value in event.channel sends notifications to default channel.
-		channelID, timestamp, err := api.PostMessage(s.Channel, slack.MsgOptionAttachments(attachment), slack.MsgOptionAsUser(true))
+	if len(event.SlackChannels) == 0 {
+		// send to all configured channel
+		event.SlackChannels = s.getAllChannels()
+	}
+	for _, channel := range event.SlackChannels {
+		channelID, timestamp, err := api.PostMessage(channel, slack.MsgOptionAttachments(attachment), slack.MsgOptionAsUser(true))
```

Comment:
This call should be async. Let's spawn go routines to send event here as well as in Mattermost